xs2a-adapter-service-api-adapter/src/main/java/de/adorsys/xs2a/adapter/http/StringUri.java

```diff
@@ -84,4 +95,19 @@ private static boolean startsWithVersion(String stringToCheck) {
     public static boolean containsProtocol(String urlToCheck) {
         return urlToCheck.contains("://");
     }
+
+    public static boolean containsQueryParam(String uri, String paramName) {
+        return getQueryParamsFromUri(uri)
+            .containsKey(paramName);
+    }
+
+    public static String appendQueryParam(String uri, String paramName, String paramValue) {
```

Comment:
should we decode param value?

Reply:
Well, we have 2 `withQuery()` methods inside this class, that add params to URI, so we either have to add encoding to all these methods as well or leave it as it is. What do you think?

Reply:
I'll create a separate ticket for that